releases/release-4.0.7.md

```diff
@@ -0,0 +1,108 @@
+    - Fix issue of plan digest is same when cop task store is different [#20076](https://github.com/pingcap/tidb/pull/20076)
+    - Fix wrong behavior for `!= any()` [#20062](https://github.com/pingcap/tidb/pull/20062)
+    - Fix issue query slow_query error when slow-log file not exist [#20051](https://github.com/pingcap/tidb/pull/20051)
+    - Fix unexpected retry region request when context cancel [#20031](https://github.com/pingcap/tidb/pull/20031)
+    - Fix issue querying cluster_slow_query time type in streaming request bug [#19943](https://github.com/pingcap/tidb/pull/19943)
+    - Fix a bug that DML using caseWhen may cause schema change [#20095](https://github.com/pingcap/tidb/pull/20095)
+    - Fix log desensitization bug in prestmt [#20048](https://github.com/pingcap/tidb/pull/20048)
+    - Fix panic tidb-server doesn't release table lock [#20020](https://github.com/pingcap/tidb/pull/20020)
+    - Fix incorrect error message of inserting enum & set [#19950](https://github.com/pingcap/tidb/pull/19950)
+    - Fix the behavior of rewrite ScalarFunction IsTure [#19903](https://github.com/pingcap/tidb/pull/19903)
+    - Fix CLUSTER_INFO system table may not work after PD is scaled-in or out [#20026](https://github.com/pingcap/tidb/pull/20026)
+    - Avoid unnecessary warnings/errors when folding constants in control expr [#19910](https://github.com/pingcap/tidb/pull/19910)
+
++ TiKV
+
+    - Fix status service shutdown when TLS handshake failed [#8649](https://github.com/tikv/tikv/pull/8649)
+    - Fix potential undefined behaviors [#7782](https://github.com/tikv/tikv/pull/7782)
+    - Fix possible panic of generating snap when doing UnsafeDestroyRange [#8681](https://github.com/tikv/tikv/pull/8681)
+
++ PD
+
+    - Fix the bug that pd might panic If some regions have no leader when `balance-region` enabled [#2994](https://github.com/pingcap/pd/pull/2994)
+    - Fix the statistical deviation of region size and region keys after region merge [#2985](https://github.com/pingcap/pd/pull/2985)
+    - Fix error hot statistic metrics [#2991](https://github.com/pingcap/pd/pull/2991)
+    - Fix miss err in `redirectSchedulerDelete` [#2974](https://github.com/pingcap/pd/pull/2974)
+
++ TiFlash
+
+    - Fix wrong result of right outer join
+
++ Tools
+
+    + Backup & Restore (BR)
+
+        - Fix a bug that caused the TiDB config changed after restoring [#509](https://github.com/pingcap/br/pull/509)
+
+    + Dumpling
+
+        - Fix the problem that dumpling fails to parse metadata when some variables are `NULL` [#150](https://github.com/pingcap/dumpling/pull/150)
+
+## Others
+
++ TiDB
+
+    - Update stats per table to avoid OOM [#20013](https://github.com/pingcap/tidb/pull/20013)
+    - Refine error message desensitization [#20004](https://github.com/pingcap/tidb/pull/20004)
+    - Accept connections from clients using connectors from MySQL 8.0 [#19959](https://github.com/pingcap/tidb/pull/19959)
```

Comment:
@lzmhhh123 Please help confirm the category(ies) of the three notes and move them to the corresponding sections.